hazelcast/src/main/java/com/hazelcast/query/impl/predicates/AbstractPredicate.java

```diff
@@ -134,6 +137,13 @@ private Comparable convert(AttributeType entryAttributeType, Comparable entryAtt
             } else if (entryAttributeType != null) {
                 return entryAttributeType.getConverter().convert(givenAttributeValue);
             } else {
+                AttributeType attrType = ReflectionHelper.getAttributeType(entryAttributeClass);
+                if (attrType != null) {
+                    TypeConverter converter = attrType.getConverter();
+                    if (converter != null) {
+                        return converter.convert(givenAttributeValue);
+                    }
+                }
```

Comment:
The passed `entryAttributeType` should be of valid type already (`ReflectionHelper` knows about it now), so looks like there is no need in this custom logic (should be handled just fine by the above `else if` branch).

Reply:
`AbstractPredicate.convert` method already receives `entryAttributeType` parameter:

https://github.com/hazelcast/hazelcast/blob/c9ac4e54145eaf28a84535e231559da59de80779/hazelcast/src/main/java/com/hazelcast/query/impl/predicates/AbstractPredicate.java#L126

In the end, its value is the one returned from `ReflectionHelper.getAttributeType`. So, AFAIU, these lines:
 https://github.com/hazelcast/hazelcast/blob/c9ac4e54145eaf28a84535e231559da59de80779/hazelcast/src/main/java/com/hazelcast/query/impl/predicates/AbstractPredicate.java#L135-L138

are doing the same as these newly introduced ones:

https://github.com/hazelcast/hazelcast/blob/c9ac4e54145eaf28a84535e231559da59de80779/hazelcast/src/main/java/com/hazelcast/query/impl/predicates/AbstractPredicate.java#L140-L146

Looks like we can just undo all the changes to `AbstractPredicate` and everything should work just fine.

Reply:
done